src/WordPress/JsonMapper/JsonMapper.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace WordPress\JsonMapper;
+
+use WordPress\JsonMapper\Evaluators\DocBlockAnnotations;
+use WordPress\JsonMapper\Evaluators\NamespaceResolver;
+use WordPress\JsonMapper\Evaluators\PropertyMapper;
+use WordPress\JsonMapper\Property\PropertyMap;
+
+class JsonMapper {
+	private $evaluators;
+
+	public function __construct() {}
+
+	private function configure_evaluators() {
+		if ( null === $this->evaluators ) {
```

Comment:
Let's do this in the constructor

Reply:
From PR description:

> Reworks Evaluators to auto-configure just before mapping, not during initialization, as was the case prior to this PR. JsonMapper used to resolve all Handlers during initiation, later checking if the resolved values are still available when mapping methods were called. This caused some code to be run during container preparation. This was altered to configure at first map_to_class call, thus no mapper-related code is run before JSON input validation succeeds.

Reply:
I'm not sure I follow. So merely setting up an array of evaluators conflicts with the container setup? That's unexpected, it sounds like lower level structures depend on higher level structures here.

Reply:
There is no conflict. It runs code it might not need. It can happen at this stage. With the magic method handlers it made some unexpected things, but now with the code streamlined it should be less of a problem.

But still - this is run always once, and I think if there is a chance JSON validation will fail and this code won't be needed there is no value in instantiating the classes at container creation.

Reply:
> But still - this is run always once, and I think if there is a chance JSON validation will fail and this code won't be needed there is no value in instantiating the classes at container creation.

It's a microoptimization that, most likely, no benchmark would measure. It's not worth additional complexity, let's simplify.

Reply:
Was also handles with https://github.com/WordPress/blueprints/pull/69#discussion_r1521329385

Optimization was not a real concern of mine. The issue was to ease debugging. When I was learning what the code does my breakpoints for JsonMapper's Handlers' internals would fire even before a single map call was made. This was partially due to use of `__invoke`. It confused me immensely, so at one point I deemed it counter-intuitive. 

With the current, simpler implementation this is a non-issue, so I see why you would look at performance to understand my reason for the change.